src/renderer/services/chain/transaction/common.ts

```diff
@@ -58,26 +59,70 @@ const sendTx$ = ({ asset, recipient, amount, memo, txType, feeOptionKey }: SendT
   }
 }
 
+const getTxByChain$ = (txHash: TxHash, chain: Chain) => {
+  // helper to create `RemoteData<ApiError, never>` observable
+  const failure$ = (msg: string) =>
+    Rx.of(
+      RD.failure({
+        errorId: ErrorId.GET_TX,
+        msg
+      })
+    )
+
+  switch (chain) {
+    case BNBChain:
+      return BNB.tx$(txHash)
+    case BTCChain:
+      return BTC.tx$(txHash)
+    case ETHChain:
+      return failure$(`tx$ has not been implemented for ETH`)
+    case THORChain:
+      return THOR.tx$(txHash)
+    case CosmosChain:
+      return failure$(`tx$ has not been implemented for Cosmos`)
+    case PolkadotChain:
+      return failure$(`tx$ has not been implemented for Polkadot`)
+  }
+}
+
 /**
  * Check if transaction has been included finally
  *
- * It's mocked currently and will be fully implemented by #755
- * @see https://github.com/thorchain/asgardex-electron/issues/755
+ * It tries to poll data every 5000 seconds and will never fail.
+ * But it stops polling by getting a valid result or by reaching maximum number (50) of requests
  *
- * @param txId Transaction hash
+ * @param txHash Transaction hash
+ * @param chain Chain
  */
-const txStatus$ = (txId: string): LiveData<ApiError, O.Option<string>> =>
-  FP.pipe(
-    Rx.of(txId),
-    RxOp.delay(2500),
-    RxOp.map(O.some),
-    RxOp.map(RD.success),
-    RxOp.catchError(() =>
-      Rx.of(
-        RD.failure<ApiError>({ errorId: ErrorId.GET_TX_STATUS, msg: 'Could load tx info' })
-      )
-    ),
+const txStatus$ = (txHash: string, chain: Chain): TxLD => {
+  // max. number of requests
+  const MAX = 50
+  // Status to do another poll or not
+  const { get$: hasResult$, set: setHasResult } = observableState(false)
+  // state of counting request
+  const { get$: count$, get: getCount, set: setCount } = observableState(0)
+  // Stream to check to stop polling or not
+  const stopInterval$ = Rx.combineLatest([hasResult$, count$]).pipe(
+    RxOp.filter(([hasResult, count]) => hasResult || count > MAX)
+  )
+
+  return FP.pipe(
+    Rx.interval(5000),
+    // Run interval as long as we don't have a valid result or MAX number of requests
+    RxOp.takeUntil(stopInterval$),
+    // count requests
+    RxOp.tap(() => setCount(getCount() + 1)),
+    RxOp.switchMap((_) => getTxByChain$(txHash, chain)),
+    liveData.map((result) => {
+      // update state to stop polling
+      setHasResult(true)
+      return result
+    }),
+    // As long as we don't reach MAX, we accept succeeded result only (but no errors)
+    // After reaching MAX we don't filter anything and will show error if its happen
+    RxOp.filter((result) => (getCount() < MAX ? RD.isSuccess(result) : true)),
```

Comment:
as it works for now i'd suggest to leave it be but add a todo-note to refactor in more "rx-way" with referring to the [official docs](https://www.learnrxjs.io/learn-rxjs/operators/error_handling/retrywhen) and `Example 2`. with such helper we can use it anywhere